### PR TITLE
[Doc] Add StarRocks Restful API Standard

### PR DESCRIPTION
**Pull request by satanson**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/http/README [hunk 1]
```diff
@@ -0,0 +1 @@
+HTTP API format should follow https://github.com/StarRocks/starrocks/tree/main/docs/developers/code-style-guides/restful-api-standard.md
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/http/README:1]:
```suggestion
HTTP API format should follow [Restful API Standard](/docs/developers/code-style-guides/restful-api-standard.md)
```

File: fe/fe-core/src/main/java/com/starrocks/http/README
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/http/README:1]: the file need to be renamed to `README.md` so the github can understand it is a markdown format text file :)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by imay — APPROVED** [review]

**Pull request merged**